### PR TITLE
Create ticket button

### PR DESCRIPTION
**Pull request by alicepotter**
## Context
In this PR, I created a 'Create ticket' button displayed on the event details page if you belong to the association. This button redirects to the 'Create ticket screen'.

## What's new

- `CreateTicket.kt` : A screen where one will be able to create a ticket for a specific event.
- A 'Create ticket' button
- Tests for the 'Create ticket' button
- `CreateTicketScreen`
- `CreateTicketTest`

## Demo
If you belong to an association, on one of their event details page you don't have the 'Become staff' button anymore but the 'Create ticket' button. 
Capture d’écran 2024-05-13 à 10 13 31

### CODE REVIEW

File: app/src/main/java/com/swent/assos/ui/screens/assoDetails/EventDetails.kt [hunk 1]
```diff
@@ -156,3 +175,41 @@ fun TopNewsBar(
 fun ConfirmButton(onConfirm: () -> Unit, text: String = "Yes") {
   Text(text = text, modifier = Modifier.clickable { onConfirm() })
 }
+
+@Composable
+fun isMember(myAssociations: List<Association>, cuurentAsso: String): Boolean {
+  var isMember = false
+  for (asso in myAssociations) {
+    if (asso.id == cuurentAsso) {
```
Comment (marinPh) [inline app/src/main/java/com/swent/assos/ui/screens/assoDetails/EventDetails.kt:183]: try using when for cognotive complexity

File: app/src/main/java/com/swent/assos/ui/screens/assoDetails/EventDetails.kt [hunk 2]
```diff
@@ -80,7 +95,11 @@ fun EventDetails(eventId: String, navigationActions: NavigationActions, assoId:
       topBar = { TopNewsBar(asso, navigationActions, event) },
       floatingActionButton = {
         if (eventId != "") {
-          JoinUsButton(onClick = { confirming = true }, text = "Become Staff")
+          if (isMember(myAssociations = myAssociations, cuurentAsso = asso.id)) {
+            CreateTicketButton(navigationActions)
+          } else {
+            JoinUsButton(onClick = { confirming = true }, text = "Become Staff")
```
Comment (marinPh) [inline app/src/main/java/com/swent/assos/ui/screens/assoDetails/EventDetails.kt:101]: try using less cognitive complexity

File: app/src/androidTest/java/com/swent/assos/EventDetailsTest.kt [hunk 3]
```diff
@@ -6,32 +6,61 @@ import androidx.compose.ui.test.onNodeWithText
...
 @RunWith(AndroidJUnit4::class)
 @HiltAndroidTest
 class EventDetailsTest : SuperTest() {
   val assoID = "02s16UZba2Bsx5opTcQb"
-  val event = Event("123456", "description", assoID, Uri.EMPTY, "assoId")
+  val event1 = Event("123456", "description", assoID, Uri.EMPTY, "assoId")
+
+  private val profileId = "dxpZJlPsqzWAmBI47qtx3jvGMHX2"
+  private val firstName = "Antoine"
+  private val lastName = "Marchand"
```
[the remark this answers is not in the document]
Comment (alicepotter) [reply app/src/androidTest/java/com/swent/assos/EventDetailsTest.kt:31]: I set in the setup that the current user is Antoine.

**Review by marinPh — APPROVED** [review]
Very good PR shows to the rest of the group how to make member dependent visibility for certain features

**Pull request merged**